src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -29,33 +29,48 @@ import {
   supportsPolygon,
   supportsThorchain,
 } from '@shapeshiftoss/hdwallet-core'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import { MetaMaskShapeShiftMultiChainHDWallet } from '@shapeshiftoss/hdwallet-shapeshift-multichain'
 import { useMemo } from 'react'
 import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { selectAccountIdsByChainId } from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 type WalletSupportsChainArgs = {
   isSnapInstalled: boolean | null
+  chainAccountIds: AccountId[] // allows dynamic chain-support detection for Ledger
   chainId: ChainId
   wallet: HDWallet | null
 }
 
 // use outside react
 export const walletSupportsChain = ({
   chainId,
+  chainAccountIds,
   wallet,
   isSnapInstalled,
 }: WalletSupportsChainArgs): boolean | null => {
   if (!wallet) return false
   const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
-  // Naming is slightly weird there, but the intent is if this evaluates to false, it acts as a short circuit
-  const shortCircuitFeatureDetection =
-    !isMetaMaskMultichainWallet || (isMetaMaskMultichainWallet && isSnapInstalled)
+  // A wallet may have feature-capabilities for a chain, but not have runtime support for it
+  // e.g MM without snaps installed, or Ledger without chain account ids (meaning the user didn't connect said chain's accounts)
+  const hasRuntimeSupport = (() => {
+    if (Boolean(isLedger(wallet) && !chainAccountIds.length)) return false
+    if (isMetaMaskMultichainWallet && !isSnapInstalled) return false
+    if (!isMetaMaskMultichainWallet) return true
+
+    // We are now sure we have runtime support for the chain.
+    // This is either a Ledger with supported chain account ids, a MM wallet with snaps installed, or
+    // any other wallet, which supports static wallet feature-detection
+    return true
+  })()
+
   switch (chainId) {
     case btcChainId:
     case bchChainId:
     case dogeChainId:
     case ltcChainId:
-      return supportsBTC(wallet) && shortCircuitFeatureDetection
+      return supportsBTC(wallet) && hasRuntimeSupport
```

Comment:
remove these `hasRuntimeSupport` conditions from the switch and just early return before the switch `if (!hasRuntimeSupport) return false`

there are implied chain support scenarios baked into the switch as it stands specific to MM, that aren't true for ledger. I can import a ledger with only thorchain accounts, and we would currently eval to true for all evm chains

Reply:
Absolute 🧠  - b328d1daf9a8cd34d593e946a19c05f34ad8737f